service/lib/agama/proxy_setup.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2023] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "uri"
+
+module Agama
+  # This class is responsible of parsing the proxy url from the kernel cmdline or configured
+  # during the boot proccess of the system writing the configuration to /etc/sysconfig/proxy
+  class ProxySetup
+    include Singleton
+    include Yast
+    include Logger
+
+    CMDLINE_PATH = "/proc/cmdline"
+    CMDLINE_MENU_CONF = "/etc/cmdline-menu.conf"
+
+    # @return [URI::Generic]
+    attr_accessor :proxy
+
+    # Constructor
+    def initialize
+      Yast.import "Proxy"
+    end
+
+    def run
+      read
+      write
+    end
+
+  private
+
+    def read
+      self.proxy = proxy_from_cmdline || proxy_from_dracut
+    end
+
+    # TODO
```

Comment:
A left-over?

Reply:
It is still not implemented, just a discussed POC with the agreement on the filename, I guess we could remove the TODO as it should work.